File: Sesion 8/Actividades/Actividad 3/Actividad 3 S8.py
class Animal:
    def hacer_sonido(self):
        return "El animal hace un sonido."

class Perro(Animal):
    def hacer_sonido(self):
        return "guau guau guau"

class Gato(Animal):
    def hacer_sonido(self):
        return "miau miau grrrr"

class Vaca(Animal):
    def hacer_sonido(self):
        return "muuu muuuu"

def main():
    animales = [Perro(), Gato(), Vaca()]

    for animal in animales:
        print(animal.hacer_sonido())

main()

File: Sesion 8/Actividades/Actividad 3/Actividad 3 S8.java
class Animal {
    public String hacerSonido() {
        return "El animal hace un sonido.";
    }
}

class Perro extends Animal {
    @Override
    public String hacerSonido() {
        return "guau guau guau";
    }
}

class Gato extends Animal {
    @Override
    public String hacerSonido() {
        return "miau miau grrrr";
    }
}

class Vaca extends Animal {
    @Override
    public String hacerSonido() {
        return "muuu muuuu";
    }
}

public class Main {
    public static void main(String[] args) {
        Animal[] animales = { new Perro(), new Gato(), new Vaca() };

        for (Animal animal : animales) {
            System.out.println(animal.hacerSonido());
        }
    }
}
